pinot-common/src/main/java/org/apache/pinot/common/utils/LogicalTableUtils.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.utils;
+
+import java.io.IOException;
+import javax.annotation.Nonnull;
+import org.apache.helix.zookeeper.datamodel.ZNRecord;
+import org.apache.pinot.spi.data.LogicalTable;
+
+
+/**
+ * Utility class which contains {@link LogicalTable} related operations.
+ */
+public class LogicalTableUtils {
+
+  private LogicalTableUtils() {
+  }
+
+  /**
+   * Fetch {@link LogicalTable} from a {@link ZNRecord}.
+   */
+  public static LogicalTable fromZNRecord(@Nonnull ZNRecord record)
+      throws IOException {
+    String tableJSON = record.getSimpleField("tableJSON");
+    return LogicalTable.fromString(tableJSON);
+  }
+
+  /**
+   * Wrap {@link LogicalTable} into a {@link ZNRecord}.
+   */
+  public static ZNRecord toZNRecord(@Nonnull LogicalTable table) {
+    ZNRecord record = new ZNRecord(table.getTableName());
+    record.setSimpleField("tableJSON", table.toSingleLineJsonString());
```

Comment:
I know chose this field name :) Is this using a pattern similar to others ? What does schema and table use ? Is `logicalTableJSON` better ?

Reply:
Schemas uses `schemaJSON` to store the JSON. Tables store the JSON form of the TableConfig. [Link](https://github.com/apache/pinot/blob/8dab53ae4efd4475cbd23b7b808fb684018a4cb0/pinot-common/src/main/java/org/apache/pinot/common/utils/config/TableConfigUtils.java#L199)

We can choose the table config approach as I found it more readable.

Table Config:
<img width="634" alt="Screenshot 2025-04-11 at 12 48 47 PM" src="https://github.com/user-attachments/assets/f919e979-e484-495f-8c4b-09fc88420463" />

Schema Config:
<img width="860" alt="Screenshot 2025-04-11 at 12 48 57 PM" src="https://github.com/user-attachments/assets/0011839f-6adc-4fb8-94fe-bc19144d67b5" />

Reply:
Updated implementation.

Logical Tables
<img width="690" alt="Screenshot 2025-04-11 at 2 49 58 PM" src="https://github.com/user-attachments/assets/d95cacd0-c185-4567-94d8-97f78ab7388b" />